wrapanapi/systems/ec2.py

```diff
@@ -253,15 +250,7 @@ def refresh(self):
         """
         Re-pull the data for this stack
         """
-        try:
-            self.raw = self._api.describe_stacks(StackName=self._uuid)['Stacks'][0]
-        except BotoServerError as error:
-            if error.status == 404:
-                raise NotFoundError('stack {}'.format(self._uuid))
-            else:
-                raise
-        except IndexError:
-            raise NotFoundError('stack {}'.format(self._uuid))
+        self.raw = self.system.get_stack(self.raw.name).raw
```

Comment:
I think this is the syntax that is causing your `maximum recursion depth` errors - you are trying to set `self.raw`, and in doing so, referencing `self.raw.name`.

Reply:
https://github.com/ManageIQ/wrapanapi/blob/ed9a2a5c81d6a0b971fea9ff0e677469f44849dc/wrapanapi/systems/ec2.py#L309 for image it works the same except it's referencing self.raw.id and it works there...I think it must be somewhere else?

Reply:
Solved with last commit.